### PR TITLE
some more phpdoc fixes

### PR DESCRIPTION
**Pull request by voku**
This change is [Reviewable](https://reviewable.io/reviews/jetbrains/phpstorm-stubs/949)

### CODE REVIEW

File: Core/Core.php [hunk 1]
```diff
@@ -762,14 +761,14 @@ function create_function ($args, $code) {}
  * @param resource $resource <p>
  * The evaluated resource handle.
  * </p>
- * @return string If the given <i>handle</i> is a resource, this function
+ * @return string|false If the given <i>handle</i> is a resource, this function
```
Comment (isfedorov) [inline Core/Core.php:764]: Hi @voku ,
Could you please clarify about this change? As I see this function always return string https://github.com/php/php-src/blob/master/Zend/zend_builtin_functions.c#L1404

**Comment by b1rdex** [comment]
AFAIK `array<key-type, value-type>` syntax is already supported in the latest EAP. So I guess it might be used in the stubs too.
Here is what I found to be working in my code:
```
list<type>
array<vtype>
array<ktype, vtype>
non-empty-list<vtype>
non-empty-array<vtype>
non-empty-array<ktype, vtype>
```

Would be nice to see some comment from the team.

**Comment by wbars** [comment]
> AFAIK `array<key-type, value-type>` syntax is already supported in the latest EAP. So I guess it might be used in the stubs too.
> Here is what I found to be working in my code:
> 
> ```
> list<type>
> array<vtype>
> array<ktype, vtype>
> non-empty-list<vtype>
> non-empty-array<vtype>
> non-empty-array<ktype, vtype>
> ```
> 
> Would be nice to see some comment from the team.

@b1rdex This syntax is provided by 3rd party tools, while phpstorm-stubs project depends only on PHP itself and PhpStorm internals.

**Comment by orklah** [comment]
> @b1rdex This syntax is provided by 3rd party tools, while phpstorm-stubs project depends only on PHP itself and PhpStorm internals.

Can we imagine psalm and phpstan plugin parsing the stubs too? The basic syntax would stay in @param/return for everyone and those with the plugins installed could enjoy more detailed stubs in @psalm/phpstan tags ?

**Comment by b1rdex** [comment]
https://blog.jetbrains.com/phpstorm/2020/10/phpstorm-2020-3-eap-2/
Paragraph "Types support" says PhpStorm supports part of the mentioned syntax.

**Pull request merged**